client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -238,6 +244,25 @@ const siteSetupFlow: Flow = {
 							apiNamespace: 'wpcom/v2',
 							apiVersion: '2',
 							formData,
+						} ).then( () => {
+							if ( ! siteId || ! isDomainOnlySite ) {
+								return;
+							}
+							// The request above also converts domain-only sites to regular sites.
+							// In this case, we need to fetch its data again to get the updated slug.
+							return new Promise( ( resolve ) => {
+								resolve( dispatch( requestSite( siteId ) ) );
+							} )
+								.catch( () => {
+									// Silently fails if we couldn't fetch the site data.
+									return;
+								} )
+								.then( ( site ) => {
+									redirectionUrl = addQueryArgs(
+										{ siteSlug: new URL( ( site as SiteDetails ).URL ).host },
+										to
+									);
+								} );
```

Comment:
I'm unsure if this is the best approach - we need to get the updated slug - which, currently, is the free WPCOM address. Without this, we still get the old slug, so the site selector would always return `null`.

Reply:
I think the _best_ approach is if we just update whatever steps need siteSlug to also allow siteId and then ensure that siteId is passed to those steps. Not sure how practical this is though.